services/provider/server/server.go

```diff
@@ -117,6 +117,10 @@ func (s *OCSProviderServer) OnboardConsumer(ctx context.Context, req *pb.Onboard
 			return nil, status.Errorf(codes.Internal, "failed to create storageConsumer %q. %v", req.ConsumerName, err)
 		}
 
+		if err == errTicketInvalidStorageQuota {
+			return nil, status.Errorf(codes.InvalidArgument, "failed to create storageConsumer %q. %v", req.ConsumerName, err)
```

Comment:
Then using the InvalidArgument code the message should describe the argument in question and not a general message.